lib/emscripten/src/syscalls/unix/host_fs.rs

```diff
@@ -640,7 +412,64 @@ pub fn ___syscall102(ctx: &mut Ctx, _which: c_int, mut varargs: VarArgs) -> c_in
     }
 }
 
-// pread
+/// select = ___syscall142
+#[allow(clippy::cast_ptr_alignment)]
+pub fn ___syscall142(ctx: &mut Ctx, _which: c_int, mut varargs: VarArgs) -> c_int {
+    debug!("emscripten::___syscall142 (newselect) {}", _which);
+
+    let nfds: i32 = varargs.get(ctx);
+    let readfds: u32 = varargs.get(ctx);
+    let writefds: u32 = varargs.get(ctx);
+    let _exceptfds: u32 = varargs.get(ctx);
+    let _timeout: i32 = varargs.get(ctx);
+    let readfds_ptr = emscripten_memory_pointer!(ctx.memory(0), readfds) as _;
+    let writefds_ptr = emscripten_memory_pointer!(ctx.memory(0), writefds) as _;
```

Comment:
Should we have asserts for the alignment?  🤔

Is it just a case of potentially being slower or are there potential security issues with things not being as we expected?

I don't have a context to know right now